rust/src/client.rs

```diff
@@ -0,0 +1,166 @@
+use crate::service::Handle;
+use crate::coordinator::RoundParameters;
+use crate::participant::{Participant, Task};
+use crate::{CoordinatorPublicKey, SeedDict, SumDict, PetError, InitError};
+use sodiumoxide::crypto::box_;
+use std::sync::Arc;
+use std::collections::HashMap;
+use std::future::Future;
+use std::pin::Pin;
+use tokio::time;
+use std::time::Duration;
+
+
+/// Client-side errors
+pub enum ClientError {
+    /// Error starting the underlying `Participant`
+    ParticipantInitErr(InitError),
+
+    /// Error from the underlying `Participant`
+    ParticipantErr(PetError),
+
+    /// Error deserialising service data
+    DeserialiseErr(bincode::Error),
+}
+
+/// A client of the federated learning service
+///
+/// `Client` is responsible for communicating with the service, deserialising
+/// its messages and delegating their processing to the underlying `Participant`.
+pub struct Client {
+    /// Handle to the federated learning `Service`
+    handle: Handle,
+
+    /// The underlying `Participant`
+    particip: Participant,
```

Comment:
it's a detail but in the rest of the codebase I think we're using the full `participant`, we should try to keep it consistent.